### PR TITLE
Comment notifications

### PR DESCRIPTION
**Pull request by iceaxeliehne**
enables comment notifications similarly to base camp. utilises module hooks

### CODE REVIEW

File: system/modules/admin/models/Comment.php [hunk 1]
```diff
@@ -49,6 +49,18 @@ public function __toString() {
...
+    
+    /*
+     * get object for comment thread
+     * return object
+     */
+    public function getParentObject() {
+        if ($this->obj_table == 'comment') {
+            return $this->w->Comment->getComment($this->obj_id)->getParentObject();
+        } else {
+            return $this->w->Comment->getObject($this->obj_table,$this->obj_id);
```
Comment (adam-buckley) [inline system/modules/admin/models/Comment.php]: The if statement is unnecessary as obj_table/id will be linking to a comment anyway

File: system/modules/admin/partials/actions/listcomments.php [hunk 2]
```diff
@@ -6,4 +6,48 @@ function listcomments(\Web $w, $params) {
...
+        $recipients_form_html .= '<input type="hidden" name="is_notifications" value="1" id="is_notifications">';
+        $recipients_form_html .= '<div id="notifications_list">';
+        $parts = array_chunk($unique_recipients, 4, true);
+        foreach ($parts as $key=>$row) {
+            
+            $recipients_form_html .= '<ul class="small-block-grid-1 medium-block-grid-' . count($row) . ' section-body">';
+            foreach ($row as $user_id => $is_notify) {
+                $user = $w->Auth->getUser($user_id);
+                if (!empty($user)) {
+                    $recipients_form_html .= '<li><label class="small-12 columns">';
```
Comment (adam-buckley) [inline system/modules/admin/partials/actions/listcomments.php]: Please avoid adding unnecessary string concatenation, lines 33 to 37 can be put into one line, use double quotes and inject variables straight in to the string, makes the code much more readable

Comment (adam-buckley) [inline system/modules/admin/partials/actions/listcomments.php]: On that, try and avoid building HTML in actions. Don't forget you can still use php functions like array_chunk and foreach loops to build javascript functionality. In the interest of getting this feature in, leave it in the action for now but please tidy up string building as per above.

File: system/modules/admin/partials/actions/listcomments.php [hunk 3]
```diff
@@ -6,4 +6,48 @@ function listcomments(\Web $w, $params) {
...
+                    $recipients_form_html .= '" class=""></label></li>';                    
+                }
+            }
+            $recipients_form_html .= '</ul>';
+        }
+        $recipients_form_html .= '</div>';
+        
+        
+        
+
```
Comment (adam-buckley) [inline system/modules/admin/partials/actions/listcomments.php]: Please clean up unnecessary whitespace and commented code if its not needed

File: system/modules/admin/models/Comment.php (hunk 1 above)
Comment (iceaxeliehne) [inline system/modules/admin/models/Comment.php]: This looks like recursive function but it is calling the same method on the parent object each time. comments can be chained together eg. reply on reply on reply etc. The function keeps going until it finds a non-comment object

**Comment by iceaxeliehne** [comment]
OK. and done.

File: system/modules/admin/models/Comment.php (hunk 1 above)
Comment (adam-buckley) [inline system/modules/admin/models/Comment.php]: Ah so it is, good catch!

**Comment by iceaxeliehne** [comment]
Adam can you please review this.

**Comment by adam-buckley** [comment]
I don't know how to add another review... but the reply link doesn't work anymore, also editing a comment shouldn't show a list of people to send an email to. We do what basecamp does and email only the initial email

**Comment by iceaxeliehne** [comment]
I know about the spelling mistake.
ready for review
